src/spatialdata/_core/query/spatial_query.py

```diff
@@ -102,6 +104,165 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
+    # intrinsic coordinate system of the points could be ('x', 'y', 'z'). The counter image of the polygon in such
+    # coordinate system could be not orthogonal to the 'z' axis; in such a case we would need to project the points to
+    # the plan in which the polygon live. Let's not handle this complexity and simply raise an error if the inverse
+    # transformation is mixing the 'z' axis with the other axes.
+    sorted_input_axes_without_c = ("x", "y", "z")[: len(input_axes_without_c)]
+    spatial_transform_bb_axes = Affine(
+        spatial_transform.to_affine_matrix(input_axes=sorted_input_axes_without_c, output_axes=("x", "y")),
+        input_axes=sorted_input_axes_without_c,
+        output_axes=("x", "y"),
+    )
+    error_message = 'The transformation is mixing the "z" axis with the other axes. This case is not supported.'
+    assert spatial_transform_bb_axes.matrix[2, 0] == 0, error_message
+    assert spatial_transform_bb_axes.matrix[2, 1] == 0, error_message
+
+    # now that we checked the above, we can restrict the transformation to the "x" and "y" axes; this will make it
+    # invertible when the points to query are 3D
+    spatial_transform_bb_axes = spatial_transform_bb_axes.to_affine(input_axes=("x", "y"), output_axes=("x", "y"))
+
+    inverse = spatial_transform_bb_axes.inverse()
+    assert isinstance(inverse, Affine)
+
+    return transform(polygon_gdf, inverse)
+
+
+def _get_axes_of_tranformation(
+    element: SpatialElement, target_coordinate_system: str
+) -> tuple[ArrayLike, tuple[str, ...], tuple[str, ...]]:
+    """
+    Get the transformation matrix and the transformation's axes (ignoring `c`).
+
+    The transformation is the one from the element's intrinsic coordinate system to the query coordinate space.
+    Note that the axes which specify the query shape are not necessarily the same as the axes that are output of the
+    transformation
+
+    Parameters
+    ----------
+    element
+        SpatialData element to be transformed.
+    target_coordinate_system
+        The target coordinate system for the transformation.
+
+    Returns
+    -------
+    m_without_c
+        The transformation from the element's intrinsic coordinate system to the query coordinate space, without the
+        "c" axis.
+    input_axes_without_c
+        The axes of the element's intrinsic coordinate system, without the "c" axis.
+    output_axes_without_c
+        The axes of the query coordinate system, without the "c" axis.
+
+    """
+    from spatialdata.transformations import get_transformation
+
+    transform_to_query_space = get_transformation(element, to_coordinate_system=target_coordinate_system)
+    assert isinstance(transform_to_query_space, BaseTransformation)
+    m = _get_affine_for_element(element, transform_to_query_space)
+    input_axes_without_c = tuple([ax for ax in m.input_axes if ax != "c"])
```

Comment:
```suggestion
    input_axes_without_c = tuple(ax for ax in m.input_axes if ax != "c")
```

Reply:
this shouldwork I think ? if yes same below

Reply:
true, replaced

Reply:
true, replaced